### PR TITLE
ci: fix wheel build

### PR DESCRIPTION
**Pull request by henryiii**

### CODE REVIEW

File: .github/workflows/step_build-wheel.yaml [hunk 1]
```diff
@@ -49,4 +49,6 @@ jobs:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
-      - uses: hynek/build-and-inspect-python-package@v2
+      - uses: hynek/build-and-inspect-python-package@v2'
```
Comment (henryiii) [inline .github/workflows/step_build-wheel.yaml]:
```suggestion
      - uses: hynek/build-and-inspect-python-package@v2
```

**Comment by henryiii** [comment]
The good news is the release is fine. Just download the artifacts manually, make sure there are no duplicates, then upload them with twine. It doesn't matter if you take the macOS 11 or 13 downloads, just don't take both.  (14 will be ARM, 11-13 was Intel). Also don't take the wheel output from build-and-inspect-python-package, only the SDist.

**Comment by LecrisUT** [comment]
I think we need a `twine check` on the wheels as well, to avoid this issue happening again. Thanks for the quick fix :)

**Comment by henryiii** [comment]
Simply trying to unzip the file would catch it, actually, since https://github.com/actions/download-artifact/issues/298 garbles the bits together instead of erroring out, making invalid zips. :) (yes, twine check should be doing this. In my experience it is _really_ limited in what it can catch, but I'd think this is one thing it could do!)

**Comment by LecrisUT** [comment]
Yey, issue resolved, just had to delete one of the artifacts and re-run the CI. Still would be good to try and catch these issues in this PR. Could `hynek/build-and-inspect-python-package` use the artifacts from `cibuildwheel`?

**Comment by LecrisUT** [comment]
I've added a simple `twine check` sanity check. @henryiii do you have anything more here or :green_circle: to merge?

**Review by LecrisUT — APPROVED** [review]

**Pull request merged**